services/swagger.js

```diff
@@ -21,10 +21,10 @@ var httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
 
 var responseModelValidationLevel;
 var polymorphicValidation;
+var refCompilation;
 
-exports.init = function (logger, config, callback) {
+exports.init = function (logger, config, refCompiler, callback) {
```

Comment:
I think that the main reason as to why I did this was so that it could use the config service and thus be configured the same way as other services (and have access to any of the swagger service configuration as needed since these two services are inherently related).